docs/google/best_practices.md

```diff
@@ -81,6 +78,20 @@ circuit_op = circuit_op.repeat(100)
 short_circuit = cirq.Circuit(circuit_op for q in qubits)
 ```
 
+When compiling circuits with `CircuitOperation`s, providing a context
+with `deep=True` will preserve the `CircuitOperation`s while
+optimizing their contents. This is useful for producing a concise,
+device-compatible circuit.
+
+```python
+ctx = cirq.TransformerContext(deep=True)
+cz_circuit = cirq.optimized_for_target_gateset(
```

Comment:
@95-martin-orion This should be `cirq.optimize_for_target_gateset(short_circuit, gateset = cirq.CZTargetGateset(), context=ctx)`. 
Differences from the current statement:
* `optimize` instead of `optimized`
* gateset should be a keyword argument.
* cirq.CZTargetGateset is a class so we need to instantiate an instance of it. 

I also think that the code snippet above demonstrating use of circuit operations for reducing serialization size is wrong. I've sent another PR doing these changes, can you please take a look?  https://github.com/quantumlib/Cirq/pull/5236